internal/backend/config/set.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"fmt"
+
+	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
+)
+
+type ComponentConfig interface {
+	Validate() error
+}
+
+type emptyComponentConfig struct{}
+
+func (emptyComponentConfig) Validate() error { return nil }
+
+var EmptyComponentConfig emptyComponentConfig
+
+type baseSet interface {
+	hasDefault() bool
+	default_() ComponentConfig
+
+	hasDev() bool
+	dev() ComponentConfig
+
+	hasTest() bool
+	test() ComponentConfig
+}
+
+type Set[T ComponentConfig] struct {
```

Comment:
IMO we can get rid of `Set` and have only one configuration. The settings of `dev`, `test` and `default` should be set from the outside.

Reply:
don't think so - we want to be able in the future, or even very soon, to list all values and not just keys per configuration mode.